src/ESLintLinter.php

```diff
@@ -132,7 +132,8 @@ protected function parseLinterOutput($path, $err, $stdout, $stderr) {
         $message = new ArcanistLintMessage();
         $message->setPath($file['filePath']);
         $message->setSeverity($this->mapSeverity($offense['severity']));
-        $message->setName(idx($offense, 'ruleId', 'unknown'));
+        $ruleId = idx($offense, 'ruleId', 'unknown');
+        $message->setName($ruleId != null ? $ruleId : 'unknown');
```

Comment:
I think we can further reduce this to:

```suggestion
        $message->setName(nonempty(idx($offense, 'ruleId'), 'unknown'));
```

... to keep it concise and to avoid having to say `'unknown'` twice.

Reply:
Agreed, that looks much better :+1: I'll patch it in, give it another test run and update this PR

Reply:
Updated, tested and working :smile: 

Btw, while trying to reproduce the bug again, I noticed that's it's not caused by regular JS syntax errors, but by Flow type syntax errors